backend/src/ml_space_lambda/dataset/lambda_functions.py

```diff
@@ -197,7 +197,6 @@ def create_dataset(event, context):
         raise Exception("Dataset headers do not match expected type and scope.")
 
     if not dataset_dao.get(scope, dataset_name):
-        s3.put_object(Bucket=env_variables["DATA_BUCKET"], Key=directory_name)
```

Comment:
Was this always broken? Why does this cause it to fail, does it just create an exception?

Reply:
This was putting an actual object at the location of the dataset path in the create function. Removing this fixes deleting files from datasets without breaking the creation of datasets.

I confirmed this by running the suite of dataset integtests and they all pass.

Reply:
This was relatively new. It was added with support for empty datasets but explicitly putting an object with a key of the "directory" resulted in an empty object existing in s3 that matched the dataset path. Normally everything in s3 has a key and there is no actual directories. For the sake of visualization `s3://bucket/my/nested/file` will be rendered as two "directories" `my` and `nested` but there is only one object in the bucket `my/nested/file`. This created an empty object with the key `my/nested/` and then when you added a file we wind up with 2 objects in the bucket `my/nested/` and `my/nested/file`. Having this extra object was the cause of the bug.

Reply:
I see so it wasn't breaking it, it just created an empty file. Thanks!